src/simtools/telescope_trigger_rates.py

```diff
@@ -0,0 +1,119 @@
+            db_config,
+        )
+    else:
+        telescope_configs = ascii_handler.collect_data_from_file(args_dict["telescope_ids"])[
+            "telescope_configs"
+        ]
+
+    for array_name, telescope_ids in telescope_configs.items():
+        _logger.info(
+            f"Processing file: {args_dict['event_data_file']} with telescope config: {array_name}"
+        )
+        histograms = SimtelIOEventHistograms(
+            args_dict["event_data_file"], array_name=array_name, telescope_list=telescope_ids
+        )
+        histograms.fill()
+
+        _calculate_trigger_rates(histograms, array_name)
+
+        if args_dict["plot_histograms"]:
+            plot_simtel_event_histograms.plot(
+                histograms.histograms,
+                output_path=io_handler.IOHandler().get_output_directory(),
+                array_name=array_name,
+            )
+
+
+def _calculate_trigger_rates(histograms, array_name):
+    """
+    Calculate trigger rates from the filled histograms.
+
+    Missing
+
+    - custom definition of energy spectra
+
+    """
+    efficiency = histograms.histograms.get("energy_eff", {}).get("histogram")
+    energy_axis = histograms.histograms.get("energy_eff", {}).get("bin_edges")
+
+    cr_spectrum = get_cosmic_ray_spectrum()
+    _logger.info(f"Cosmic ray spectrum: {cr_spectrum}")
+    e_min = energy_axis[:-1] * u.TeV
+    e_max = energy_axis[1:] * u.TeV
+    cr_rates = (
+        np.array(
+            [
+                cr_spectrum.integrate_energy(e1, e2).decompose(bases=[u.s, u.cm, u.sr]).value
+                for e1, e2 in zip(e_min, e_max)
+            ]
+        )
+        * histograms.file_info["scatter_area"].to("cm2").value
+        * histograms.file_info["solid_angle"].to("sr").value
+        * u.Hz
+    )
+    trigger_rates = efficiency * cr_rates
+    trigger_rate = np.sum(trigger_rates, axis=0)
+
+    _logger.info(f"Scatter area from MC: {histograms.file_info['scatter_area'].to('m2')}")
+    _logger.info(f"Solid angle from MC: {histograms.file_info['solid_angle']}")
+    _logger.info(f"Trigger rate for {array_name} array: {trigger_rate.to('Hz')}")
+
+    histograms.histograms["cr_rates_mc"] = histograms.get_histogram_definition(
+        histogram=cr_rates.value,
+        bin_edges=energy_axis,
+        title="Cosmic Ray Rates (MC)",
+        axis_titles=["Energy (TeV)", "Cosmic Ray Rate (Hz)"],
+        plot_scales={"x": "log", "y": "log"},
+    )
+    histograms.histograms["trigger_rates"] = histograms.get_histogram_definition(
+        histogram=trigger_rates.value,
+        bin_edges=energy_axis,
+        title="Trigger Rates (MC)",
+        axis_titles=["Energy (TeV)", "Trigger Rate (Hz)"],
+        plot_scales={"x": "log", "y": "log"},
+    )
+
+    return cr_rates, trigger_rates, trigger_rate
+
+
+def get_cosmic_ray_spectrum():
+    """
+    Return the cosmic ray spectrum.
+
+    To be extended in future to read a larger variety of spectra.
```

Comment:
A very important comment, this was about to be my suggestion for the review. In particular, the option to use the "all particle spectra" in order to calculate the "real" data rate for CTAO. Is there an issue open to make sure it's not forgotten?

Reply:
Related to #931, so not forgotten (it would already be there, I was waiting for this PR).